Source/CompilationDVDView.cpp

```diff
@@ -0,0 +1,341 @@
+	if ((node.InitCheck() != B_OK) || !node.IsDirectory())
+		return;
+
+	fDirPath->SetTo(&entry);
+	if (strcmp("VIDEO_TS", fDirPath->Leaf()) == 0)
+		fDirPath->GetParent(fDirPath);
+	else {
+		BPath testPath(fDirPath->Path());
+		testPath.Append("VIDEO_TS");
+		get_ref_for_path(testPath.Path(), &ref);
+		if (!BEntry(&ref).Exists()) {
+			fBurnerInfoBox->SetLabel(B_TRANSLATE_COMMENT(
+				"Didn't find a VIDEO_TS folder", "Status notification"));
+			return;
+		}
+	}
+
+	fPathView->SetText(fDirPath->Path());
+	fImageButton->SetEnabled(true);
+	fBurnButton->SetEnabled(false);
+	fBurnerInfoBox->SetLabel(B_TRANSLATE_COMMENT("Build the DVD UDF image",
+		"Status notification"));
+
+	BPath audioTSPath(fDirPath->Path());
+	audioTSPath.Append("AUDIO_TS");
+	create_directory(audioTSPath.Path(), 0777);
+}
+
+
+void
+CompilationDVDView::_BurnerOutput(BMessage* message)
+{
+	BString DVD;
+
+	if (message->FindString("line", &DVD) == B_OK) {
+		DVD << "\n";
+		fBurnerInfoTextView->Insert(DVD.String());
+		fBurnerInfoTextView->ScrollTo(0.0, 2000.0);
+	}
+	int32 code = -1;
+	if ((message->FindInt32("thread_exit", &code) == B_OK) && (step == 1)) {
+		BString infoText(fBurnerInfoTextView->Text());
+		if (infoText.FindFirst(
+			"mkisofs: Unable to make a DVD-Video image.\n") != B_ERROR) {
+			fBurnerInfoBox->SetLabel(B_TRANSLATE_COMMENT(
+				"Unable to make a DVD-Video image",
+				"Status notification"));
+			fBurnButton->SetEnabled(false);
+		} else {
+			fBurnerInfoBox->SetLabel(B_TRANSLATE_COMMENT("Burn the disc",
+				"Status notification"));
+			fImageButton->SetEnabled(false);
+			fBurnButton->SetEnabled(true);
+		}
+		step = 0;
+
+	} else if ((message->FindInt32("thread_exit", &code) == B_OK)
+			&& (step == 2)) {
+		fBurnerInfoBox->SetLabel(B_TRANSLATE_COMMENT(
+			"Burning complete. Burn another disc?", "Status notification"));
+		fDVDButton->SetEnabled(true);
+		fImageButton->SetEnabled(false);
+		fBurnButton->SetEnabled(true);
+
+		step = 0;
+	}
+}
+
+
+#pragma mark -- Public Methods --
+
+
+void
+CompilationDVDView::BuildISO()
+{
+	if (fDirPath->Path() == NULL) {
+		(new BAlert("ChooseDirectoryFirstAlert",
+			B_TRANSLATE("First choose DVD folder to burn."),
+			B_TRANSLATE("OK")))->Go();
+		return;
+	}
+
+	if (fBurnerThread != NULL)
+		delete fBurnerThread;
+
+	fBurnerInfoTextView->SetText(NULL);
+	fBurnerInfoBox->SetLabel(B_TRANSLATE_COMMENT(
+		"Building in progress" B_UTF8_ELLIPSIS, "Status notification"));
+	fBurnerThread = new CommandThread(NULL,
+		new BInvoker(new BMessage(kBurnerMessage), this));
+
+	if (find_directory(B_SYSTEM_CACHE_DIRECTORY, fImagePath) != B_OK)
+		return;
+
+	status_t ret = fImagePath->Append("burnitnow_dvd.iso");
+	if (ret == B_OK) {
+		step = 1;	// flag we're building ISO
+
+		fBurnerThread->AddArgument("mkisofs")
+		->AddArgument("-V")
+		->AddArgument(fDirPath->Leaf())
+		->AddArgument("-dvd-video")
+		->AddArgument("-o")
+		->AddArgument(fImagePath->Path())
+		->AddArgument(fDirPath->Path())
+		->Run();
```

Comment:
Indentation looks suspicious here.

Reply:
True. Here and in all other "tab" files, the ->AddArgument()s should be once indented wrt fBurnerThread->AddArgument().